### PR TITLE
(maint) Improving pa_matchbatch.sh

### PR DESCRIPTION
**Pull request by smcclellan**
This adds `REPO` for improved handling of running pa_matchbatch.sh for
private repos.

### CODE REVIEW

File: pa_matchbatch.sh [hunk 1]
```diff
@@ -203,7 +220,7 @@ for currentItem in ${repoRevMap}; do
 	#    [[ "${ignored_repos}" =~ ${foss_name} ]]
 	# will not work when $ignored_repos contains e.g. puppet-agent
 	# and $foss_name is puppet.
-	if containsElement "${ignored_repos}" "${foss_name}"; then
+	if containsElement "${ignored_repos} .git" "${foss_name}"; then
```
Comment (ekinanp) [inline pa_matchbatch.sh]: Why is there `.git` here?

File: pa_matchbatch.sh [hunk 2]
```diff
@@ -2,9 +2,21 @@
...
+# - WORKSPACE: Where the dependencies should be downloaded, e.g. `temp`.
+# - IGNORE_FOR: Which repos should be ignored (space-separated).
+# - OVERRIDE_PATH: A path to a txt file that will override versions.
+#
+# Deprecated arguments:
+# - PUPPET_AGENT_URL: This will still be read, but `REPO` is preferred.
+
 # where to find the agent
-puppet_agent_repo="git@github.com:puppetlabs/puppet-agent.git"
+REPO=${REPO:-puppet-agent}
```
Comment (ekinanp) [inline pa_matchbatch.sh:16]: Should probably update the top-level `README` with this argument.

File: pa_matchbatch.sh (hunk 1 above)
Comment (scottsolowild) [reply pa_matchbatch.sh]: The .git directory was being included, and throwing errors, for me.

File: pa_matchbatch.sh (hunk 2 above)
Comment (scottsolowild) [reply pa_matchbatch.sh:16]: Added.

File: pa_matchbatch.sh (hunk 1 above)
Comment (scottsolowild) [reply pa_matchbatch.sh]: I'm not seeing the same errors now, so going to remove this.

**Review by ekinanp — APPROVED** [review]

**Pull request merged**